plugins/janus_nosip.c

```diff
@@ -1899,7 +1878,13 @@ static int janus_nosip_allocate_local_ports(janus_nosip_session *session) {
 		}
 	}
 	/* We need this to quickly interrupt the poll when it's time to update a session or wrap up */
-	pipe(session->media.pipefd);
+	if(!update) {
+		pipe(session->media.pipefd);
+	}
+	/* Let relay_thread to accommodate the changes */
+	if(update) {
+		session->media.updated = TRUE;
```

Comment:
This is currently already done at the end of `janus_nosip_sdp_process`, which also uses `pipefd` to wake the relay thread. We do it there because it's when the peer sends us an SDP, and so we use the potentially new port to reconnect the sockets.

My feeling is that you only need this here as well if the local SDP is an answer, as in that case there wouldn't be another SDP in response to trigger `janus_nosip_sdp_process`. As such, the `if(update)` should become something like if(update && answer)`, and you'll need to add the pipe wakeup here too. Does this make sense in your scenario?

Reply:
You right, the `janus_nosip_sdp_process` wakeup thread on next message.
I think just a
`if(update) {
    wakeup()
}`
would be the correct behaviour. Please, confirm.